topostats/grainstats.py

```diff
@@ -241,28 +241,31 @@ def calculate_stats(self) -> Dict:
             # from pixel units to nanometres.
             # Removed formatting, better to keep accurate until the end, including in CSV, then shorten display
             stats = {
-                "centre_x": centre_x * self.pixel_to_nanometre_scaling,
-                "centre_y": centre_y * self.pixel_to_nanometre_scaling,
-                "radius_min": radius_stats["min"] * self.pixel_to_nanometre_scaling,
-                "radius_max": radius_stats["max"] * self.pixel_to_nanometre_scaling,
-                "radius_mean": radius_stats["mean"] * self.pixel_to_nanometre_scaling,
-                "radius_median": radius_stats["median"] * self.pixel_to_nanometre_scaling,
-                "height_min": np.nanmin(masked_grain_image),
-                "height_max": np.nanmax(masked_grain_image),
-                "height_median": np.nanmedian(masked_grain_image),
-                "height_mean": np.nanmean(masked_grain_image),
-                "volume": np.nansum(masked_grain_image) * self.pixel_to_nanometre_scaling**2,
-                "area": region.area * self.pixel_to_nanometre_scaling**2,
-                "area_cartesian_bbox": region.area_bbox * self.pixel_to_nanometre_scaling**2,
-                "smallest_bounding_width": smallest_bounding_width * self.pixel_to_nanometre_scaling,
-                "smallest_bounding_length": smallest_bounding_length * self.pixel_to_nanometre_scaling,
+                "centre_x": centre_x * self.pixel_to_nanometre_scaling * 1e-9,
```

Comment:
Moved the multiplication to up-front. 👍 

Also removed one of the extraneous plots, however I think that your suggestion of plotting the grains might require some extra values being added to `grainstats` DataFrame as in order to do the plotting we need various values pertaining to bounding box of the grains, namely `minr, minc, maxr, maxc = region.bbox`. I think the grain mask obtained and used internally to `calculate_grainstats` will need to be passed out too, as we plot sections of that. 

Do you have any suggestions about this?

Reply:
I was thinking that grainstats should either have a property or return a dictionary of all the data that is required to plot grains. This can then optionally be processed (via dictionary comprehension) to use the classes in `plottingfuncs.py` to plot and save grain level images if required.

My reasoning being that plotting is a distinct action from calculating statistics on grains. If not treated as separate plotting should perhaps be done at the grain detection stage rather than under the statistics calculation, from memory the bounding boxes and other things that isolate grains are all derived via scikit-image's morphology.

Let me know what you think and happy to talk this through in person.

Reply:
Thanks! I'll give it a go 👍